File: Leetcode 3345 Smallest Divisible Digit Product I.go
// 2026/08/06

func smallestNumber(n int, t int) int {
    var tp int
    for i:=n; i < 101; i++ {
        tp = 1
        for _, j := range strconv.Itoa(i) {
            tp *= int(j - '0')
        }
        if tp % t == 0{
            return i
        }
    }
    return 0
}

File: Leetcode 3345 Smallest Divisible Digit Product I.py
# 2026/08/06

class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        for i in range(n, 101):
            tp = 1
            for num in str(i):
                tp *= int(num)
            if tp % t == 0:
                return i
